app/uat_actions/uat_actions/generate_sample_manifest.rb

```diff
@@ -97,30 +100,19 @@ def create_samples(sample_manifest)
 
   private
 
-  def purpose
-    Purpose.find_by!(name: tube_purpose_name)
-  end
+  # Validates that the plate purpose exists for the selected plate purpose name.
+  #
+  # @return [void]
+  def validate_plate_purpose_exists
+    return if plate_purpose_name.blank?
+    return if PlatePurpose.exists?(name: plate_purpose_name)
 
-  def study
-    @study ||=
-      Study.create_with(
-        state: 'active',
-        study_metadata_attributes: {
-          data_access_group: 'dag',
-          study_type: UatActions::StaticRecords.study_type,
-          faculty_sponsor: UatActions::StaticRecords.faculty_sponsor,
-          data_release_study_type: UatActions::StaticRecords.data_release_study_type,
-          study_description: 'A study generated for UAT',
-          contaminated_human_dna: 'No',
-          contains_human_dna: 'No',
-          commercially_available: 'No',
-          program: UatActions::StaticRecords.program
-        }
-      ).find_or_create_by!(name: study_name)
+    message = format(ERROR_PLATE_PURPOSE_DOES_NOT_EXIST, plate_purpose_name)
+    errors.add(:plate_purpose_name, message)
   end
 
-  def supplier
-    @supplier ||= Supplier.find_or_create_by!(name: supplier_name)
+  def purpose
+    Purpose.find_by!(name: tube_purpose_name)
```

Comment:
I wondered about this tube_purpose_name, it says plate purpose in other places, it says tube purpose in the UI, does this manifest only work for tubes? or poor method and variable naming?

Reply:
In this case it is just the variable naming. Form field definition adds an attr_accessor to the uat action class with that name.